### ISSUE TITLE
Map clusters don't render when using all the data.

### ISSUE THREAD

**Issue by rgreinho** [scrapd/scrapdviz#97]
Issue Type
----------

- Bug report

Current Behavior
----------------

If I choose to access all the data in the data picker, the map does not show any cluster. However it works fine if I select the data year by year.

Expected Behavior
-----------------

We should be able to visualize all the markers on the map.

**Comment by rgreinho** [comment]
Assigned to @AndrewGibson27.

**Comment by rgreinho** [comment]
Digging a little bit more, and comparing to the archive data set, it look like if there is a **SINGLE** malformed pair of coordinates, the map won't render anything.

The clustering function should validate (maybe with a regex), and discard invalid data.

**Comment by rgreinho** [comment]
On the APD view, if you choose `12/05/2017` as the starting date, the map will work.

**Comment by rgreinho** [comment]
@AndrewGibson27 I found the problem for this one! 🎉  

It happens that we have invalid or undefined coordinates in the data sets. `react-mapbox-gl` requires valid coordinates in order to create a marker. If the marker creation process fails, no markers are displayed on the map at all. 

It looks like we only have a few incorrect locations (I found 16 so far). That's why depending on the date boundaries that are selected some times it works (all coordinates within the range are valid), and some times it does not (at least one set of coordinates is invalid).

I'll add a fix tonight to filter out the invalid coordinates from the marker creation process.
